tests/include/test/psa_crypto_helpers.h

```diff
@@ -36,19 +36,29 @@
 const char *mbedtls_test_helper_is_psa_leaking( void );
 
 /** Check that no PSA Crypto key slots are in use.
+ *
+ * If any slots are in use, mark the current test as failed and jump to
+ * the exit label. This is equivalent to
+ * `TEST_ASSERT( ! mbedtls_test_helper_is_psa_leaking( ) )`
+ * but with a more informative message.
  */
-#define ASSERT_PSA_PRISTINE( )                                  \
-    TEST_ASSERT( ! mbedtls_test_helper_is_psa_leaking( ) )
+#define ASSERT_PSA_PRISTINE( )                                          \
+    do                                                                  \
+    {                                                                   \
```

Comment:
Why not:
```c
        const char *msg = mbedtls_test_helper_is_psa_leaking( );        \       
        if( msg != NULL )                                               \       
        {                                                               \       
            test_fail( msg, __LINE__, __FILE__ );                       \       
            goto exit;                                                  \       
        }                                                               \
```
That way we don't need to add the PSA specific `test_fail_if_psa_leaking()` function in helpers.function (with a similar change in PSA_DONE() below.

Reply:
This is a macro. I prefer to avoid defining a variable locally in a macro because it's unhygienic (what if the same variable name was also used in the surrounding code?). I'd do it if I have to, but I prefer to put the variable in a function, as I did here.

Reply:
Looking at this a bit more closely, why are PSA_DONE() and ASSERT_PSA_PRISTINE() not defined in helpers.function? They both rely on test_fail() which is part of the unit test case execution machinery thus defining them in ./tests/include just seems to make things more difficult.

Reply:
`test_fail` should eventually moved out of `helpers.function`. Now that `tests/src` exists, I think all of `helpers.function` should move there, and possibly most of `*_test.function` as well. But that's way out of scope here.

Originally, I created `psa_crypto_helpers.h` partly to avoid bloating `helpers.function` too much, and partly because Mbed TLS and Mbed Crypto were separate projects but `helpers.function` had to be kept in synch between the two so it was easier to keep crypto-specific code in the same file as much as possible. Thankfully that second reason doesn't apply, but the first one still does.

Reply:
ok. It seems it was not clear to me where we were heading regarding helpers.function and tests/src. A bit clearer now, thanks.